src/main/java/org/lwjglx/opengl/Display.java

```diff
@@ -164,34 +164,7 @@ public void invoke(long window, int key, int scancode, int action, int mods) {
                             KeyEvent.getKeyText(KeyCodes.lwjglToAwt(KeyCodes.glfwToLwjgl(key))),
                             (key >= 32 && key < 127) ? ((char) key) : '?');
                 }
-                latestEventKey = key;
-                if (Config.IME_ENABLED) {
-                    if (Config.IME_F12_TOGGLE && key == GLFW_KEY_F12 && action == 0) {
-                        imeOn = !imeOn;
-                    }
-                    if (Config.IME_SYS_TOGGLE
-                            && (Keyboard.isKeyDown(Keyboard.KEY_LCONTROL) || Keyboard.isKeyDown(Keyboard.KEY_RCONTROL)
-                                    || Keyboard.isKeyDown(Keyboard.KEY_LWIN))) {
-                        if (key == GLFW_KEY_SPACE || key == GLFW_KEY_LEFT_SHIFT || key == GLFW_KEY_RIGHT_SHIFT) {
-                            imeOn = !imeOn;
-                            return;
-                        }
-                    }
-                    if (imeOn) {
-                        if (key > 256) {
-                            Keyboard.addGlfwKeyEvent(window, key, scancode, action, mods);
-                        }
-                    } else {
-                        Keyboard.addGlfwKeyEvent(window, key, scancode, action, mods);
-                    }
-                } else {
-                    Keyboard.addGlfwKeyEvent(window, key, scancode, action, mods);
-                }
-                // Ctrl should generate ASCII modifier keys (0x01-0x1F), glfw does not give use char events for this
-                if ((mods & GLFW_MOD_CONTROL) != 0 && key >= GLFW_KEY_A && key <= GLFW_KEY_Z) {
```

Comment:
This is still needed, textboxes rely on these to handle ctrl+c/ctrl+v combinations

Reply:
Vanilla handles ctrl c/v by calling isKeyDown(), which mod rely on this exactly?

Reply:
Vanilla, see GuiTextField textboxKeyTyped: it switches on special char values 1, 3, 22, 24